tests/unit/test_sagemaker_steps.py

```diff
@@ -292,6 +293,84 @@ def test_training_step_creation_with_placeholders(pca_estimator):
             'TrialComponentDisplayName': 'Training'
         },
         tags=DEFAULT_TAGS,
+        hyperparameters=execution_input['HyperParameters']
```

Comment:
clarifying: can any of these other properties be expressed as placeholders? - if so, let's plug the gap to prevent future regressions. Ideally we have a test that covers all placeholder properties and not just some. 

That was the issue that made this bug easy to introduce, so lets it harder to introduce a bug by strengthening tests where possible.

Reply:
All properties that were documented as being compatible with Placeholders are included in this test

We have an open feature request (#117) to support Placeholders for all parameters for Sagemaker steps - these will be added to the test as they are made Placeholder compatible